### PR TITLE
ELPP-2376 Test pipeline

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Test should fail, replicating what I had on the Vagrant VM; both PHPUnit and the smoke tests should now be checking the correct setup.

**Comment by giorgiosironi** [comment]
3 checks will be made in sequence by this build:
- code style
- PHPUnit
- smoke test via curl

Now it's stuck on the first one, get this green and now we can create the production server.

**Comment by garyttierney** [comment]
@giorgiosironi is there a way to make it run an additional test command?  I'd like to run `./gradlew -d backend check` and get a test report (NUnit/JUnit compatible report) on the backend code too, if possible.

**Comment by giorgiosironi** [comment]
Yes, the `project_tests.sh` script contains all the steps, so you can add checks there in the order you prefer (`smoke_tests.sh` is separate because it's run on all environments after deploy rather than only in `ci`).

File: src/App/Kernel.php [hunk 1]
```diff
@@ -77,24 +78,32 @@ public function dependencies(Application $app)
         $app['schematron.controller'] = function (Application $app) {
             return new DefaultController($app['schematron.backend_client']);
         };
+        $app['ping.controller'] = function() {
```
Comment (thewilkybarkid) [inline src/App/Kernel.php]: Can be replaced with `$app->register(new PingControllerProvider());` after updating `elife/ping`.

File: composer.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
         "symfony/var-dumper": "^3.1",
         "silex/web-profiler": "^2.0",
         "jms/serializer": "^1.4",
-        "kevinrob/guzzle-cache-middleware": "^1.5"
+        "kevinrob/guzzle-cache-middleware": "^1.5",
+        "elife/ping": "^1.0@dev"
```
Comment (thewilkybarkid) [inline composer.json]: Now has a stable release.

File: tests/bootstrap [hunk 3]
```diff
@@ -9,7 +9,6 @@
 namespace {
     require_once __DIR__ . '/../vendor/autoload.php';
     require_once __DIR__ . '/../vendor/elife/api-sdk/test/TestCase.php';
-    require_once __DIR__ . '/../vendor/elife/api-sdk/test/PuliAwareTestCase.php';
```
Comment (giorgiosironi) [inline tests/bootstrap]: I think most of these can be removed?

Comment (giorgiosironi) [reply tests/bootstrap]: all of them apart from `vendor/autoload.php`

**Review by garyttierney — APPROVED** [review]

**Comment by giorgiosironi** [comment]
Will merge, and create `schematron-validator--prod` to deploy it.

**Pull request merged**